### ISSUE TITLE
can display desktop just like windows sandbox, or proxy windows sandbox's fileSystem?

### ISSUE THREAD

**Issue by LY1806620741** [sandboxie-plus/Sandboxie#4118]
### Is your feature request related to a problem or use case?

I want sandboxie simulate my system to create develop environment. As we know, clean develop tools file in disk alway be hard. And many soft use disk C:// too large,  alway no free space.

### Describe the solution you'd like

right click sandboxie environment then click open a new destop, and show all soft lnk on the desktop what we ceate, just like windows sandbox system function.

### Describe alternatives you've considered

config windows sandbox config *.wsb to use sandboxie fileSystem

**Comment by love-code-yeyixiao** [comment]
Good idea.Maybe we could mount a pre-prepared system image into memory and redirect all read operations? (But that would probably require you to have at least 30GB of physical memory)

**Comment by coltsfootrock** [comment]
Please don't consider doing this with Sbie.
Putting resources into mounting system images etc, for me, just goes beyond what Sbie should be doing.

**Comment by love-code-yeyixiao** [comment]
No no no,this is possible in fact,we have already an additional plugin “ImDisk”,so it maybe will be very easy to mount a system image file,and we have memory sandbox,so only thing we need to do is to mount a image instead of create an empty memory area like nowdays.Sandboxed Desktop and Sandboxed Session these two features are working.



---Original---
From: ***@***.***&gt;
Date: Sun, Jul 28, 2024 00:38 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [sandboxie-plus/Sandboxie] can display desktop just like windowssandbox, or proxy windows sandbox's fileSystem? (Issue #4118)




 
Please don't consider doing this with Sbie.
 Putting resources into mounting system images etc, for me, just goes beyond what Sbie should be doing.

**Comment by love-code-yeyixiao** [comment]
After discussion with the project maintainer, the basic final solution is as follows: the user needs to prepare a clean virtual disk file that has been installed with the Windows operating system by himself (it doesn't matter if it's not clean. And as long as your virtual machine software, such as HyperV supports VMDX files, then you can now install the system in the virtual machine and then mount it to Sandboxie), after the user independently chooses to mount the virtual disk to Sandboxie, you can run the program normally. At this time, when the sandboxed process tries to read files and registry, Sandboxie will first redirect its read request to the sandboxed folder, and if there is no copy in the folder, it will redirect it to the previous virtual disk file, if there is no file in the virtual disk, then it will directly skip the host lookup, and directly return to the "File not found! ". For write requests, we redirect them directly to the sandbox folder.

This simply simulates a clean file and registry environment for the sandboxed application. To complete the requests in this issue, there is another work-in-progress (on hold) feature that can help with this. That is the "sandboxed desktop", which you can learn about briefly at this link and its accompanying Youtbe video: "https://www.wilderssecurity.com/threads/sandboxed-desktop-new-feature-work-in-progress.454168/"
Simply put, if you use another shell project that can be sandboxed as a sandboxed desktop shell, you will get the experience you need (at this stage, we can't, and won't, run Explorer.exe as a sandboxed shell).
